support/Python/Schedule.py

```diff
@@ -415,16 +415,21 @@ def schedule(
     if segments_dir and all_segments:
         # Make sure we're continuing the last checkpoint of the last segment.
         # This requirement can be relaxed in the future if needed.
-        assert (
-            from_checkpoint
-            and Path(from_checkpoint).resolve()
-            == all_segments[-1].checkpoints[-1].path.resolve()
-        ), (
-            "You're not continuing from the previous segment's "
-            "last checkpoint. This is technically possible, "
-            "but probably wrong. This assert safeguards against "
-            "inconsistent usage."
-        )
+        try:
+            assert from_checkpoint
+            last_checkpoint = all_segments[-1].checkpoints[-1]
+            assert (
+                Path(from_checkpoint).resolve()
+                == last_checkpoint.path.resolve()
+            )
+        except (AssertionError, IndexError) as err:
+            raise AssertionError(
+                "You're not continuing from the previous segment's last"
+                " checkpoint. This is technically possible, but probably"
+                " wrong. This assert safeguards against inconsistent usage."
+                " Either run in a different directory, or use"
+                " '--from-checkpoint' to continue from the last checkpoint."
```

Comment:
I split this error message into three, but kept the "last segment" vs "last checkpoint" together. The error prints the expected path of the checkpoint, which includes both the segment and the checkpoint, so I see no point in splitting it into separate errors messages.

Reply:
I still think the third error message you have isn't clear.

```
You're not continuing from the previous segment's last checkpoint
(Segment001/Checkpoints/Checkpoint000).
```

I agree that this is a true statement and that it tells you what you should be doing (i.e. continuing from this specific checkpoint which is the last checkpoint in the latest segment). However, it's confusing because it doesn't tell me what I did wrong. Did I specify a segment other than the latest one? Or did I specify the correct segment, but not the last checkpoint in the latest segment?

And to me, that's confusing because I'm going to go back to the (incorrect) command I used and look for the offending option. If I specified an incorrect segment, that's what I'll be looking for. If it's an incorrect checkpoint, then I'll be looking for that. But I have to know what I'm looking for in the first place and your error message doesn't tell me that.

Also, I didn't find this line very helpful.

```
Either run in a different directory, or use '--from[-last]-checkpoint'
to continue from the last checkpoint.
```

Again, what I should be doing is dependent on whether I specified the incorrect segment or checkpoint. If I specified the correct segment but an incorrect checkpoint, the suggestion is simple:

```
Use '--from-last-checkpoint {last_segment}' or '--from-checkpoint {last_checkpoint}' to
continue from the last checkpoint.
```

However, if I specified an incorrect segment it's a bit trickier because of potentially corrupted segments. Did I actually want to continue from the segment I specified? If so, then I need to remove all segments after that one. Or did I just accidentally specify the wrong segment and I meant to continue from the latest one? For both of these I think a message like this would be helpful:

```
You are not continuing from the latest segment {last_segment}. If you want to continue
from the last checkpoint in the segment you specified ({specified_segment}), you must
remove all the segments after that one [list segments to remove?]. Otherwise, specify
'--from-last-checkpoint {last_segment}' to continue from the latest segment.
```

So overall I think this should be the order of the `asserts`:

1. assert from_checkpoint, "Found segments. Use '--from-last-segment {last_segment}' to continue from the last checkpoint in {last_segment}"
2. assert we are in last segment. If not, use above message.
3. assert current segment (which is the latest segment) has checkpoints, "Oh no, no checkpoints to continue from"
4. assert continuing from last checkpoint in this segment (the latest segment). if not use above message.

I know this is a lot of logic and similar error messages whereas yours is less logic and combined messages. But I think the differences in the message that are tailored to the specific errors are very clear, will help users a lot, and make the experience of using the scheduler a lot smoother. What do you think?

Reply:
See fixup

Reply:
Sorry, but I still think these error messages will be confusing for users. However, I've held up this PR long enough and the messages are already decent-ish so I think it's better to just get this merged and I can do a follow-up PR.